beetsplug/kodinfo.py

```diff
@@ -0,0 +1,448 @@
+
+        except (ValueError, TypeError):
+            # catch simplejson.decoder.JSONDecodeError and load emptydata
+            data2 = json.loads(emptyalbum)["album"][0]
+
+        out_data = ("{0};{1};{2};{3};{4};{5};{6};{7};{8};{9};{10};{11};{12};"
+                    "{13};{14}".format((data[3]),
+                                       (data[8]),
+                                       (data[0]),
+                                       (data[4]),
+                                       (data2["strStyle"]) or '',
+                                       (data2["strMood"]) or '',
+                                       (data2["strTheme"]) or '',
+                                       (comp),
+                                       (data2["strReview"]) or '',
+                                       (data[7]),
+                                       (rel_date),
+                                       (data[6]),
+                                       (data2["intScore"]) or '',
+                                       (date[0]),
+                                       (data2["strAlbumThumb"]) or ''))
+        return list(out_data.split(';'))
+
+
+def album_tracks(albumid):
+    """Get album's tracks from beets libary."""
+    trackdata = []
+    for item in lib.items(albumid):
+        row = item.track, item.mb_trackid, item.length, item.title
+        duration = time.strftime("%M:%S", time.gmtime(row[2]))
+        trackdata.append(list(tuple([row[0], duration, row[1], row[3]])))
+        tracklist = (sorted(trackdata))  # sort list by track number
+    return tracklist
+
+
+def kodi_path():
+    """From kodi itself get the music library path."""
+    """Useful for shared libraries, in order to get nfs or samba paths."""
+    try:
+        auth = str.encode(
+            '%s:%s' %
+            (config['kodi']['user'],
+             config['kodi']['pwd']))
+        authorization = b'Basic ' + base64.b64encode(auth)
+        headers = {
+            'Content-Type': 'application/json',
+            'Authorization': authorization}
+        url = "http://{0}:{1}/jsonrpc".format(
+            config['kodi']['host'], config['kodi']['port'])
+        data = {"jsonrpc": "2.0",
+                "method": "Files.GetSources",
+                "params": {"media": "music"},
+                "id": 1}
+        json_data = json.dumps(data).encode('utf-8')
+        request = Request(url, json_data, headers)
+        result = json.load(urlopen(request))
+        _kodi_path = result['result']['sources'][0]['file']
+        return _kodi_path
+    except (requests.exceptions.RequestException, ValueError, TypeError):
+        return ''
+
+
+def album_path(albumid):
+    """Get album path."""
+    out_data = []
+    for album in lib.albums(albumid):
+        album_path = album.path.decode("utf-8")
+        root = str(config["directory"])
+        length = int(len(root) + 1)
+        kodi_lib_path = kodi_path() + album_path[length:]
```

Comment:
Not every path in the beets database starts with the `directory` path; it's possible to have out-of-library files. So `album_path[length:]` could create a malformed path. (As above, it would be helpful to know why this "transplanting" of paths is necessary.)